src/components/WeekDayPicker/weekDay.js

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useUniqueIdentifier } from '../../libs/hooks';
+import getWeekDayLabel from './helpers/getWeekDayLabel';
+import { StyledInput, StyledWeekDayContainer, StyledWeekDayLabel } from './styled';
+
+const WeekDay = React.forwardRef((props, ref) => {
+    const {
+        name,
+        value,
+        locale,
+        isChecked,
+        disabled,
+        required,
+        readOnly,
+        multiple,
+        error,
+        onChange,
+    } = props;
+
+    const weekDayId = useUniqueIdentifier('week-day');
+    const inputType = multiple ? 'checkbox' : 'radio';
```

Comment:
We have to optimize as much as possible. Use the hook useMemo () here.

Reply:
Here I think there is no need to use useMemo, keep in mind that useMemo is a function that does more things inside it and it is called every renders too, we should use this optimization when:
1. Referential equality is needed to keep thing at the same reference like objects, arrays or functions.
2. Computationally expensive calculations.
Here is a good guide:
https://kentcdodds.com/blog/usememo-and-usecallback